pkg/xds/envoy/envoy.go

```diff
@@ -140,17 +146,32 @@ func CreatePassThroughCluster(clusterName string) *v2.Cluster {
 	}
 }
 
-func CreateOutboundListener(ctx xds_context.Context, listenerName string, address string, port uint32, clusterName string, virtual bool, sourceService string, destinationService string, backends []*v1alpha1.LoggingBackend, proxy *core_xds.Proxy) (*v2.Listener, error) {
+func CreateOutboundListener(ctx xds_context.Context, listenerName string, address string, port uint32, statsName string, clusters []ClusterInfo, virtual bool, sourceService string, destinationService string, backends []*v1alpha1.LoggingBackend, proxy *core_xds.Proxy) (*v2.Listener, error) {
```

Comment:
We've got 11 arguments to this method. Can we group it somehow?

Just an idea:
```
type OutboundListener struct {
	name string
	address string
	port uint32
	statsName string
	virtual bool
	clusters []ClusterInfo
}
func CreateOutboundListener(ctx xds_context.Context, listener OutboundListener, backends []*v1alpha1.LoggingBackend, proxy *core_xds.Proxy) (*v2.Listener, error) {
	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
	destinationService := listener.statsName
	...
}
```

What do you think?

Reply:
`type CreateOutboundListenerOpts struct { ... }` is a standard solution in that case.

However, we need a bigger refactoring/restructuring of Envoy config generating logic.

So, I'll leave it as is for now.

Reply:
ok, let's just try to do it eventually so the code won't get out of control